src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -320,6 +320,106 @@ void QgsExpressionCalculatorLocatorFilter::triggerResult( const QgsLocatorResult
 {
   QApplication::clipboard()->setText( result.userData.toString() );
 }
+
+// SettingsLocatorFilter
+//
+QgsSettingsLocatorFilter::QgsSettingsLocatorFilter( QObject *parent )
+  : QgsLocatorFilter( parent )
+{}
+
+QgsSettingsLocatorFilter *QgsSettingsLocatorFilter::clone() const
+{
+  return new QgsSettingsLocatorFilter();
+}
+
+void QgsSettingsLocatorFilter::fetchResults( const QString &string, const QgsLocatorContext &, QgsFeedback *feedback )
+{
+  QMap<QString, SettingsHandle> filteredSettingHandlesMap;
+  QMap<QString, QString> *optionsPagesMap = QgisApp::instance()->getOptionsPagesMap();
+
+  for ( QString op : optionsPagesMap->keys() )
```

Comment:
Probably better to use a const iterator, you can make a const copy of `optionsPagesMap->keys()` and iterate over it (same for other for loops below)

Reply:
Actually, for performance you should do:

    for ( auto optionsIt = optionsPagesMap->constBegin(); optionsIt != optionsPagesMap->constEnd(); ++optionsIt )

In general you should avoid iterating using QMap/QHash::keys() - because internally that causes an iteration over the whole map and building a new list for the key values. Better to iterate directly over the original map and avoid the duplicate iteration & list construction.

Note this is especially bad code, kill it if you ever see it:

```
   QMap< X, Y > myMap;
   for ( const X& key : myMap.keys() )
   {
      Y val = myMap.value( key );
      ...
   }
```

This should instead be:

 ```
   QMap< X, Y > myMap;
   for ( auto it = myMap.constBegin(); it != myMap.constEnd(); ++it )
   {
      X key =  it.key();
      Y val = it.value();
      ...
   }
```

Because:
- single iteration, no duplicate iteration to build key list and unnecessary construction of an extra list
- no slow map/hash lookup on every key value

/qt lesson over :smile: